src/couch_changes.erl

```diff
@@ -336,6 +343,16 @@ get_doc_ids(_) ->
     throw({bad_request, no_doc_ids_provided}).
 
 
+get_selector({json_req, {Props}}) ->
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(#httpd{method='POST'}=Req) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    {Props} = couch_httpd:json_body_obj(Req),
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(_) ->
+    throw({bad_request, invalid_selector}).
```

Comment:
In case of bad selector type - yes, but in case of GET request it's just missed. May be add one more clause about:
```
get_selector({json_req, _}) -> throw({bad_request, invalid_selector});
```

And the this clause turns into something more friendly:
```
get_selector(_) -> throw({bad_request, "Selector must be specified in POST payload"}).
```

What do you think?